volatility3/framework/symbols/linux/extensions/elf.py

```diff
@@ -72,7 +88,10 @@ def is_valid(self):
         """
         Determine whether it is a valid object
         """
-        return self._type_prefix is not None and self._hdr is not None
+        if self._valid_magic:
+            return self._type_prefix is not None and self._hdr is not None
```

Comment:
Rather than rereading the flag, needing an extra variable and so on, I'd just check `self._hdr.e_ident` to verify it's correct?  Carry state across when the value is already present in part of the state seems like an unnecessary waste?